bin/create-scripts/block.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+const path = require( 'path' );
+const webpack = require( 'webpack' );
+
+const __rootDir = path.resolve( __dirname, '../../' );
+const entryPath = path.resolve( process.argv[ 2 ] );
+const sourceDir = path.dirname( entryPath );
+const outputDir = path.join( sourceDir, 'build' );
+const blockName = path.basename( path.dirname( entryPath ) );
+
+const baseConfig = require( path.join( __rootDir, 'webpack.config.js' ) );
+
+const config = {
+	...baseConfig,
+	...{
+		mode: 'production',
+		entry: entryPath,
+		externals: {
+			...baseConfig.externals,
+			wp: 'wp',
+		},
+		optimization: {
+			splitChunks: false,
+		},
+		output: {
+			path: outputDir,
+			filename: `blocks-${ blockName }.js`,
+			libraryTarget: 'window',
+			library: `blocks-${ blockName }`,
+		},
+		module: {
+			...baseConfig.module,
+			rules: [
+				...baseConfig.module.rules,
+				{
+					test: /\.scss$/,
+					use: [
+						'style-loader',
+						'css-loader',
+						{
+							loader: 'sass-loader',
+							options: {
+								includePaths: [
+									path.join( __rootDir, 'assets', 'stylesheets' ),
+									path.join( __rootDir, 'assets', 'stylesheets', 'shared' ),
+									path.join( __rootDir, 'assets', 'stylesheets', 'shared', 'mixins' ),
+								],
+							},
+						},
+					],
+				},
+			],
+		},
+	},
+};
+
+const compiler = webpack( config );
+
+compiler.run( ( error, stats ) => console.log( stats.toString() ) );
```

Comment:
I know this is very early in the process and probably subject to change/maturation but you're using `webpack` while we're using `babel` + `sass` directly to build our modules in Gutenberg. I'm not arguing about one approach being better than the other but my question is whether you think there's value in aligning practices between Calypso and WordPress Core in this area (build tools) as well? Could it be one of the goals of this project to be able to move things easily between Calypso and Core?

As an aside, we can make changes to our build process in Gutenberg if this approach is better, but changes like these need to go through a discussion in #core-js meetings for Gutenberg.

Reply:
Sounds reasonable to me. Agreed, the scope of the project is different and aligning with Core is not a priority especially at the beginning. 

I'd say the most important thing is probably to take this opportunity to build decoupled "modules/directories/packages" because it's easier to adapt small independent packages one by one than rewriting a monolith. 

So yeah, it's probably better to focus on the MVP without this extra constraint but since we're doing the exact same things (building modules and React code in both worlds) keeping it in mind to avoid having big boundaries between the two is important longer term IMO.